### ISSUE TITLE
PEP 3105 Make print a function in twisted/python

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8400]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8400|
|Type|enhancement|
|Created|2016-06-03 10:47:53Z|
|Branch|https://github.com/twisted/twisted/tree/8400-printfunc-python|

This ticket was split out of [ticket:5812] which was too big to review.

PEP 3105 ( ​​​https://www.python.org/dev/peps/pep-3105/ ) **print** deprecates has changed from a statement to a function.

In Python 2.6 and Python 2.7, the default is print as a statement

```
print "Hello"
```

In Python 3.x, the above example is an error, since only print as a function is available.

To consistently use **print** as a function
on Python 2.6, Python 2.7, and Python 3.x, this should be done:

```
from __future__ import print_function

print("Hello")
```

<details><summary>Searchable metadata</summary>

```
trac-id__8400 8400
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__8400_printfunc_python 8400-printfunc-python
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1464950873793880 1464950873793880
changetime__1465727636476802 1465727636476802
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

See:
https://github.com/twisted/twisted/pull/116

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)|@hawkowl set owner to @hawkowl|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_5"></a>|@hawkowl commented|
|-|-|

LGTM, just waiting for builders to go green...

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_6"></a>|@hawkowl set status to `closed`|
|-|-|

In changeset 7eb0221e320fdd35a96e8ca36c0afea76e3200d8
```
#!CommitTicketReference repository="" revision="7eb0221e320fdd35a96e8ca36c0afea76e3200d8"
Merge 8400-printfunc-python: Make print a function in twisted.python

Author: rodrigc
Reviewers: alex, hawkowl
Fixes: #8400
```
